src/logistration/tests/RegistrationPage.test.jsx

```diff
@@ -49,6 +50,233 @@ describe('./RegistrationPage.js', () => {
     jest.clearAllMocks();
   });
 
+  it('should show error message on invalid email', () => {
+    const validationMessage = 'Enter a valid email address that contains at least 3 characters.';
+    store = mockStore({
+      ...store,
+      logistration: {
+        ...store.logistration,
+        registrationResult: {
+          success: false,
+          redirectUrl: '',
+        },
+        registrationError: null,
+        formData: {
+          fields: [{
+            label: 'I agree to the Your Platform Name Here <a href="/honor" rel="noopener" target="_blank">Honor Code</a>',
+            name: 'honor_code',
+            type: 'checkbox',
+            errorMessages: {
+              required: 'You must agree to the Your Platform Name Here Honor Code',
+            },
+          }],
+        },
+      },
+    });
```

Comment:
spread operator should be called on `initialState` and not on the store (it is a MockStore object)

```suggestion
    store = mockStore({
      ...initialState,
      logistration: {
        ...initialState.logistration,
        registrationError: null,
        formData: {
          fields: [{
            label: 'I agree to the Your Platform Name Here <a href="/honor" rel="noopener" target="_blank">Honor Code</a>',
            name: 'honor_code',
            type: 'checkbox',
            errorMessages: {
              required: 'You must agree to the Your Platform Name Here Honor Code',
            },
          }],
        },
      },
    });
```

Reply:
update is required in all tests

Reply:
The rest i would refactor it

Reply:
done